### DISCUSSION TITLE
xCalendar on .net MAUI not responding to any user input/click and wrong order of days when using starting week day as Monday

### DISCUSSION THREAD

**Discussion by EugenM88** [ME-MarvinE/XCalendar#131] [Q&A] [answered]
Greetings.

First of all i would like to say thank you very much for making such an amazing and much needed control for .net MAUI.
It's amazing how much you can do with it and how many customizations you can achieve with such controls, far surpassing even WinUI native calendar control!

Now to get to the issue:
The calendar

The calendar days work fine and whenever i do a programmatic change in selected dates:
xcCalendar.SelectedDates = new XCalendar.Core.Collections.ObservableRangeCollection<DateTime>() { DateTime.Today, DateTime.Today.AddDays(1) };
Such dates are reflected automatically as expected in the UI.

However clicking anything in the android emulator does not allow the control to respond to anything, be it clicking the left or right arrow in the title bar or clicking individual date circles, its like the control is completely disabled from the user input.

Also, not sure if this is intended but the calendar "title bar" renders the entire text in white, while the background itself is also white, making everything invisible unless i change the entire control background color, im pretty sure im missing something but i couldnt find for the life of me what, even after following the startup guide for the maui project line by line.

![image](https://user-images.githubusercontent.com/43586673/236290893-717cab15-380a-4ca1-bde1-b7839c860666.png)

If i change the back color, then you can see the hidden controls

![image](https://user-images.githubusercontent.com/43586673/236291630-4a0c6ae5-5caa-46bf-9e35-4f98cb935840.png)

Thank you very much in advance for your help and once again thank you for such an amazing control!

**Comment by EugenM88** [comment]
Hello, it turns out nothing is wrong with the control, its just my implementation is far from sufficient to get a working calendar, in order for it to respond to any input and be able to change some specific colors i needed to use more coding from the control repository example, namely:
SwipableCalendarExamplePage.xaml and SelectionExamplePage.xaml, its cs classes and viewmodels.
![image](https://user-images.githubusercontent.com/43586673/236492748-6bb87c58-b988-4722-9878-2df1639cca33.png)

It semms to be working almost fine now, however i've found 1 tiny problem still remains, namely, when i change the weekday display order using code. for example:
`Calendar.DayNamesOrder = new XCalendar.Core.Collections.ObservableRangeCollection<DayOfWeek>() {DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday, DayOfWeek.Saturday, DayOfWeek.Sunday };`

The days numbers are out of place. Am i changing the order incorrectly or is there something else i am probably missing?

**Comment by EugenM88** [comment]
And finally, to fix this issue, instead of using DayNamesOrder property, you should rather leave it alone and use this instead for changing the first day of week:

`StartOfWeek = DayOfWeek.Monday,`

![image](https://user-images.githubusercontent.com/43586673/236529598-1218e47c-e072-4149-bbd7-47a64f93df4c.png)


Now everything shows up as it should!

I hope this little newb adventure will help someone who might stumble on the same issues i did.

Best Regards!

**Reply by ME-MarvinE** [reply, marked as answer]
Wow, nice job! Sorry I didn't reply sooner, I only just got a notification about this discussion. Im glad you were able to figure it out with the resources provided in the repo.

I don't know why the title bar showed up as white. By default it should be red.
